src/lazy/encoding.rs

```diff
@@ -60,6 +60,8 @@ pub trait Encoding: Encoder + Decoder {
     ) -> IonResult<W> {
         Self::default_write_config().encode_all_to(output, values)
     }
+
+    fn encoding(&self) -> IonEncoding;
```

Comment:
🗺️ Types implementing the `Encoding` trait now must report which variant of the `IonEncoding` enum they correspond to.